MdeModulePkg/Universal/SetupBrowserDxe/IfrParse.c

```diff
@@ -438,8 +438,9 @@ CreateStorage (
   EFI_GUID         *StorageGuid;
   CHAR8            *StorageName;
 
-  UnicodeString = NULL;
-  StorageName   = NULL;
+  UnicodeString  = NULL;
+  StorageName    = NULL;
+  BrowserStorage = NULL; // MU_CHANGE - Initialize variable that might not be updated due to error checking

```

Comment:
This is a good catch, it looks like we need to do the same for Storage, as if we fail on UnicodeString being NULL, we would attempt to check if Storage->ConfigRequest is NULL, free it if not, then do the same with Storage. So can you also update Storage to be initialized to NULL and update the error checking to check if Storage is NULL first, then if Storage->ConfigRequest is NULL, free it, then free Storage?

Reply:
Sorry, misread the error flow. We shouldn't need to initialize Storage, but we should change the order of the error checking to check Storage before Storage->ConfigRequest (and only check inside Storage if it is non-null).

Reply:
```c
   if (Storage != NULL) {
     if (Storage->ConfigRequest != NULL) {
      FreePool (Storage->ConfigRequest);
    }
    FreePool (Storage);
  }
```

Reply:
Thank you. Changed the checking flow.